beaker_kernel/lib/context.py

```diff
@@ -98,6 +99,14 @@ def preview(self) -> Callable[[], Awaitable[Any]] | None:
         if preview_func and inspect.iscoroutinefunction(preview_func):
             return preview_func
 
+    @property
+    def fetch_kernel_state(self) -> Callable[[], Awaitable[Any]] | None:
+        state_func = getattr(self, self.kernel_state_function_name, None)
```

Comment:
This, combined with `kernel_state_function_name: str = "fetch_kernel_state"` above creates a endless recursive loop if `fetch_kernel_state` is not overridden.
The overridden version in the default context is not a property.
When overriding, it's best to keep the types/definitions consistent.